launchable/utils/session.py

```diff
@@ -14,47 +12,33 @@ def _session_file_dir() -> Path:
     return Path(os.environ.get(SESSION_DIR_KEY) or DEFAULT_SESSION_DIR).expanduser()
 
 
-def _session_file_path(build_name: str) -> Path:
-    return _session_file_dir() / (hashlib.sha1("{}:{}".format(build_name, _get_session_id()).encode()).hexdigest() + ".txt")
-
-
-def _get_session_id() -> str:
-    # CircleCI changes unix session id each steps, so set non change variable
-    # https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables
-    if os.environ.get("CIRCLECI") is not None:
-        id = os.environ.get("CIRCLE_WORKFLOW_ID")
-        if id is not None:
-            return id
-        Logger().warning("CIRCLECI environment variable is set but not CIRCLE_WORKFLOW_ID")
-
-    # Jenkins pipeline also launches every process as a new session, so better to scope this to a particular build.
-    # For freestyle job types that do not do this, session IDs do not distinguish different builds, so that is a problem, too.
-    #
-    # See: https://github.com/jenkinsci/lib-durable-task/blob/6e020747205cb5aca5af757bc0d2a302c42bdb79/src/cmd/bash/durable_task_monitor.go#L41
-    # See: https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables
-    if os.environ.get("JENKINS_URL") is not None:
-        id = os.environ.get("BUILD_URL")
-        if id is not None:
-            return id
-        Logger().warning("JENKINS_URL environment variable is set but not BUILD_URL")
-
-    if sys.platform == "win32":
-        import wmi  # type: ignore
-        c = wmi.WMI()
-        wql = "Associators of {{Win32_Process='{}'}} Where Resultclass = Win32_LogonSession Assocclass = Win32_SessionProcess".format(
-            os.getpid())
-        res = c.query(wql)
-        return str(res[0].LogonId)
-    else:
-        return str(os.getsid(os.getpid()))
+def _session_file_path() -> Path:
+    return _session_file_dir() / ".launchable"
+
+
+def _parse_session_file(text: str):
+    try:
+        build, session = text.split("#")
+        _, build_name = build.split("=")
+        _, session_id = session.split("=")
+
+        return build_name, session_id
+    except Exception as e:
+        raise Exception("Can't parse session file: {}".format(text)) from e
 
 
 def read_session(build_name: str) -> Optional[str]:
-    f = _session_file_path(build_name)
+    f = _session_file_path()
     try:
         if not f.exists():
             return None
-        return f.read_text()
+        build, session_id = _parse_session_file(f.read_text())
+
+        if build != build_name:
+            raise Exception("Build name is different between input:{} and saved:{}. Previous job might be failed.\nPlease confirm previous job result and run after removing {}".format(
```

Comment:
How is this error message eventually get shown to users?

CLI error handling generally falls into two categories; if an error is anticipated, as in this case, then the error message should not be shown with a stack trace, which distracts users. If an error is unanticipated, then we want to see the entire stack trace so that we can understand where the problem happened and how.

My typical design pattern is to designate an exception type for anticipated errors, so that the code that's catching it can use the type as a hint to decide what to do. When I see a raw `Exception` being thrown like this, I don't think we can make the distinction above, which degrade the user experience.

This is our CLI design deficiency that goes beyond this PR, but we might as well fix it here (or in the next PR, I suppose)